tests/integration/test_data_store_mgr.py

```diff
@@ -303,3 +309,81 @@ def test_delta_task_prerequisite(harness):
         p.satisfied
         for t in schd.data_store_mgr.updated[TASK_PROXIES].values()
         for p in t.prerequisites})
+
+
+@pytest.fixture
+def ghost_job_db(request: pytest.FixtureRequest):
+    """Provide workflow DB for "ghost job" test."""
+    db_dump = (
+        request.path.parent / 'fixture-data' / 'ghost-job.sqlite3'
+    ).read_text()
+
+    def _ghost_job_db(srv_dir_path):
+        srv_dir_path = Path(srv_dir_path)
+        srv_dir_path.mkdir(parents=True, exist_ok=True)
+        con = sqlite3.connect(srv_dir_path / WorkflowFiles.Service.DB)
+        cur = con.cursor()
+        cur.executescript(db_dump)
+        con.close()
+
+    return _ghost_job_db
+
+
+async def test_ghost_job(
+    flow, one_conf, scheduler, run, ghost_job_db, db_select, gql_query
+):
+    """Test restarting when there is a "ghost job" in the DB
+    (has a submit_time in task_jobs table but no submit_exit_time or run_time)
+    """
+    reg = flow(one_conf)
+    ghost_job_db(get_workflow_srv_dir(reg))
+    schd: Scheduler = scheduler(reg, paused_start=True)
+
+    db_columns = ('cycle', 'name', 'submit_num')
+
+    async with run(schd):
+        client = WorkflowRuntimeClient(reg)
+        await asyncio.sleep(1)  # yields control to main loop
+
+        # There will be 1 ghost job in DB:
+        assert db_select(
+            schd, False, DAO.TABLE_TASK_JOBS, *db_columns
+        ) == [
+            ('1', 'one', 1)
+        ]
+        # Get submit timestamp for that job:
+        orig_submit_time = db_select(
+            schd, False, DAO.TABLE_TASK_JOBS, 'time_submit'
+        )[0][0]
+        # Ghost job should not be in data store:
+        assert await gql_query(client, '''
+            jobs {
+                cyclePoint, name, submitNum
+            }
+        ''') == {'jobs': []}
+
+        schd.resume_workflow()
+        await asyncio.sleep(1)  # yields control to main loop
+
+        # Job should now be in data store:
+        # assert await gql_query(client, '''
+        #     jobs {
+        #         cyclePoint, name, submitNum
+        #     }
+        # ''') == {
+        #     'jobs': [{
+        #         'cyclePoint': '1',
+        #         'name': 'one',
+        #         'submitNum': 1
+        #     }]
+        # }
```

Comment:
I couldn't get this bit to work. Any idea how to test that the job is in the data store?

Reply:
Yeah, we should be able to access it via the `schd`